editor/src/core/tailwind/tailwind-class-list-utils.spec.ts

```diff
@@ -0,0 +1,193 @@
+import type { TailwindClassParserResult } from './tailwind-class-list-utils'
+import {
+  addNewClasses,
+  getClassListFromParsedClassList,
+  getParsedClassList,
+  removeClasses,
+  updateExistingClasses,
+} from './tailwind-class-list-utils'
+
+describe('tailwind class list utils', () => {
+  describe('class list parsing / writing', () => {
+    function classNameAstSlice(ast: TailwindClassParserResult) {
+      switch (ast.type) {
+        case 'unparsed':
+          return ast
+        case 'parsed':
+          const slice: Record<string, unknown> = {
+            type: ast.type,
+            property: ast.ast.property,
+            value: ast.ast.value,
+          }
+          if (ast.ast.negative) {
+            slice.negative = true
+          }
+          return slice
+      }
+    }
+    it('can parse the class list', () => {
+      const classList = getParsedClassList(
+        'p-4 m-4 lg:w-4 hover:text-red-100 -top-4 flex flex-row fancyButton',
+        null,
+      )
+      expect(classList.map(classNameAstSlice)).toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "property": "padding",
+            "type": "parsed",
+            "value": "1rem",
+          },
+          Object {
+            "property": "margin",
+            "type": "parsed",
+            "value": "1rem",
+          },
+          Object {
+            "property": "width",
+            "type": "parsed",
+            "value": "1rem",
+          },
```

Comment:
this doesn't capture the `lg:` prefix. is that going to lead to data loss when we manipulate the entries?

Reply:
updated the tests here https://github.com/concrete-utopia/utopia/pull/6506/commits/97f400a7e10d410a10a7f6d5237b540fc8700885 to make it explicit that we do preserve the prefixes